ipi/engine/forces.py

```diff
@@ -1331,16 +1377,54 @@ def pot_combine(self):
         return rp
 
     def extra_combine(self):
-        """Obtains the extra string for each forcefield for each bead."""
+        """Combines the extra dictionaries for each forcefield for each bead."""
 
         self.queue()
-        rp = [listDict() for k in range(self.nforces)]
+
+        re = {}
         for k in range(self.nforces):
-            # "expand" to the total number of beads the potentials from the
-            # contracted one
-            for b in range(self.nbeads):
-                rp[k].append(self.mforces[k].extras[b])
-        return rp
+            # combines the extras from the different force components
+            for e, v in self.mforces[k].extras.items():
+                if e in self.mforces[k].force_extras:
+                    # extras that are tagged as force_extras are treated exactly as if they were an energy/force/stress
+                    v = (
+                        self.mforces[k].weight
+                        * self.mforces[k].mts_weights.sum()
+                        * self.mrpc[k].b2tob1(v)
+                    )
+                    if e in re:
```

Comment:
Uhh.... I thought we had agreed that silently summing everything was not a good idea. Why is this now being done regardless?

Reply:
So, the rationale here is that quantities that are "promoted" to force_extras status are treated as if they were forces. This means they can be contracted, combined, summed and weighted. I can see more cases in which this is useful than cases in which it is a problem: if you don't want your extras to be summed, you can (a) not include them in force_extras or (b) give them a different name in each forcefield.

Reply:
So essentially the default is "don't touch the extras". If you activate the force_extras machinery, then they get summed and resampled.